File: go-client-sdk/pkg/models/shared/codatdatacontractsdatasetsageddebtoroutstanding.go
package shared

type CodatDataContractsDatasetsAgedDebtorOutstanding struct {
	AgedCurrencyOutstanding []CodatDataContractsDatasetsAgedCurrencyOutstanding `json:"agedCurrencyOutstanding,omitempty"`
	CustomerID              *string                                             `json:"customerId,omitempty"`
	CustomerName            *string                                             `json:"customerName,omitempty"`
}

File: typescript-client-sdk/src/sdk/models/shared/codatdatacontractsdatasetsageddebtoroutstanding.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { CodatDataContractsDatasetsAgedCurrencyOutstanding } from "./codatdatacontractsdatasetsagedcurrencyoutstanding";
import { Expose, Type } from "class-transformer";


export class CodatDataContractsDatasetsAgedDebtorOutstanding extends SpeakeasyBase {
  @SpeakeasyMetadata({ elemType: CodatDataContractsDatasetsAgedCurrencyOutstanding })
  @Expose({ name: "agedCurrencyOutstanding" })
  @Type(() => CodatDataContractsDatasetsAgedCurrencyOutstanding)
  agedCurrencyOutstanding?: CodatDataContractsDatasetsAgedCurrencyOutstanding[];

  @SpeakeasyMetadata()
  @Expose({ name: "customerId" })
  customerId?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "customerName" })
  customerName?: string;
}